samples/bluetooth/central_smp_client/src/main.c

```diff
@@ -318,18 +353,26 @@ static int send_smp_echo(struct bt_dfu_smp *dfu_smp,
 			 const char *string)
 {
 	static struct smp_buffer smp_cmd;
-	CborEncoder cbor, cbor_map;
+	zcbor_state_t zse[CBOR_ENCODER_STATE_NUM];
 	size_t payload_len;
-	struct cbor_buf_writer writer;
 
-	cbor_buf_writer_init(&writer, smp_cmd.payload, sizeof(smp_cmd.payload));
-	cbor_encoder_init(&cbor, &writer.enc, 0);
-	cbor_encoder_create_map(&cbor, &cbor_map, 1);
-	cbor_encode_text_stringz(&cbor_map, "d");
-	cbor_encode_text_stringz(&cbor_map, string);
-	cbor_encoder_close_container(&cbor, &cbor_map);
+	zcbor_new_encode_state(zse, ARRAY_SIZE(zse), smp_cmd.payload,
+			       sizeof(smp_cmd.payload), 0);
+
+	/* Stop encoding on the error. */
+	zse->constant_state->stop_on_error = true;
+
+	zcbor_map_start_encode(zse, CBOR_MAP_MAX_ELEMENT_CNT);
+	zcbor_tstr_put_lit(zse, "d");
+	zcbor_tstr_put_term(zse, string);
+	zcbor_map_end_encode(zse, CBOR_MAP_MAX_ELEMENT_CNT);
+
+	if (!zcbor_check_error(zse)) {
```

Comment:
The check is reverse.

Reply:
Is it handle correctly here ? Because I want here to return an error when encoding process will fail